Lib/pdb.py

```diff
@@ -1437,13 +1436,19 @@ def do_display(self, arg):
         Without expression, list all display expressions for the current frame.
         """
         if not arg:
-            self.message('Currently displaying:')
-            for item in self.displaying.get(self.curframe, {}).items():
-                self.message('%s: %r' % item)
+            if self.displaying:
+                self.message('Currently displaying:')
+                for item in self.displaying.get(self.curframe, {}).items():
+                    self.message('%s: %r' % item)
+            else:
+                self.message('No expression is being displayed')
         else:
-            val = self._getval_except(arg)
-            self.displaying.setdefault(self.curframe, {})[arg] = val
-            self.message('display %s: %r' % (arg, val))
+            val, exc = self._getval_except(arg)
+            if isinstance(exc, SyntaxError):
+                self.message(f'Unable to display "{arg}": SyntaxError')
```

Comment:
Consider using `{arg!r}` instead of `"{arg}"`.

e.g.  for `arg = '"'`

`print(f'"{arg}"')` outputs `"""`
`print(f'"{arg!r}"')` outputs `'"'`

for `arg = '\\'`

`print(f'"{arg}"')` outputs `"\"`
`print(f'"{arg!r}"')` outputs `'\\'`

Reply:
Or just simply use `'%r' % arg`